internal/tracing/helper.go

```diff
@@ -28,8 +29,12 @@ func StartTracing(ctx context.Context, spanName string, tracingEnabled bool, att
 	return ctx, span
 }
 
-func EndTracing(span trace.Span) {
+func EndTracing(span trace.Span, err error) {
 	if span != nil {
+		span.RecordError(err)
+		if err != nil {
+			span.SetStatus(codes.Error, err.Error())
+		}
 		span.End()
```

Comment:
Shouldn't `span.RecordError(err)` be inside `if err != nil`?

Reply:
Hello @kuba-4chain 

Yes, it could be but it will not be a problem because: 

> ... If this span is not being recorded or err is nil then this method does nothing.

https://pkg.go.dev/go.opentelemetry.io/otel/trace@v1.31.0#Span.RecordError

Reply:
Moved it inside